jobloop/jobloop.go

```diff
@@ -32,6 +32,9 @@ type Job interface {
 	// ProcessOne finds and executes exactly one task. If no task is available to
 	// be executed, `sql.ErrNoRows` is returned.
 	ProcessOne() error
+	// Process finds and executes a given amount of tasks. If not enough tasks are available to
+	// be executed, `sql.ErrNoRows` is returned. If any error is encountered, processing stops early.
+	Process(int) error
```

Comment:
Since the implementation is identical for each Job, this should rather be a free-standing function to avoid code duplication:

```
func Process(j Job, count int) error { ... }
```

Also, the name is quite generic. How about `ProcessMany` or `ProcessMultiple` as a counterpart to `ProcessOne`?